test/Mono.Linker.Tests.Cases/DataFlow/LocalDataFlow.cs

```diff
@@ -221,13 +221,14 @@ public static void TestBranchGoto ()
 			return;
 		}
 
+		// https://github.com/dotnet/linker/issues/2550
```

Comment:
Would you mind moving the call to this method in `Main` to the section with "extraneous warnings"? And the issue link could go there as well.